app.js

```diff
@@ -249,44 +277,292 @@ const app = Vue.createApp({
+            maxHamming_eq4 = this.maxHammingData['eq4'][String(eq4_val)]*step
+
+
+            if (!isNaN(available_distance_eq1)){
+                n_existing_lower=n_existing_lower+1
+                percent_to_next_eq1_hamming = (current_hamming_distance_eq1)/maxHamming_eq1
+                //can be nan if divided by zero
+                if(isNaN(percent_to_next_eq1_hamming)){
+                    percent_to_next_eq1_hamming=0
+                }
+                
+                normalized_hamming_eq1 = available_distance_eq1*percent_to_next_eq1_hamming
+                
+            }
+
+            if (!isNaN(available_distance_eq2)){
+                n_existing_lower=n_existing_lower+1
+                percent_to_next_eq2_hamming = (current_hamming_distance_eq2)/maxHamming_eq2
+                if(isNaN(percent_to_next_eq2_hamming)){
+                    percent_to_next_eq2_hamming=0
+                }
+                
+                normalized_hamming_eq2 = available_distance_eq2*percent_to_next_eq2_hamming
+            }
+
+            if (!isNaN(available_distance_eq3eq6)){
+                n_existing_lower=n_existing_lower+1
+                percent_to_next_eq3eq6_hamming = (current_hamming_distance_eq3eq6)/maxHamming_eq3eq6
+                if(isNaN(percent_to_next_eq3eq6_hamming)){
+                    percent_to_next_eq3eq6_hamming=0
+                }
+                
+                normalized_hamming_eq3eq6 = available_distance_eq3eq6*percent_to_next_eq3eq6_hamming
+            }
+
+            if (!isNaN(available_distance_eq4)){
+                n_existing_lower=n_existing_lower+1
+                percent_to_next_eq4_hamming = (current_hamming_distance_eq4)/maxHamming_eq4
+                if(isNaN(percent_to_next_eq4_hamming)){
+                    percent_to_next_eq4_hamming=0
+                }
+                
+                normalized_hamming_eq4 = available_distance_eq4*percent_to_next_eq4_hamming
+            }
+
+            if (!isNaN(available_distance_eq5)){
+                //for eq5 is always 0 the percentage
+                n_existing_lower=n_existing_lower+1
+                percent_to_next_eq5_hamming = 0
+                if(isNaN(percent_to_next_eq5_hamming)){
+                    percent_to_next_eq5_hamming=0
+                }
+                
+                normalized_hamming_eq5 = available_distance_eq5*percent_to_next_eq5_hamming
+            }
+
+            if (n_existing_lower==0) {
+                mean_distance = 0
+            } else { //sometimes we need to go up but there is nothing there, or down but there is nothing there so it's a change of 0.
+                mean_distance = (normalized_hamming_eq1+normalized_hamming_eq2+normalized_hamming_eq3eq6+normalized_hamming_eq4+normalized_hamming_eq5)/n_existing_lower
+            }
+            value = parseFloat(value) - parseFloat(mean_distance);
```

Comment:
We absolutely need to avoid floating point arithmetic, except last step. All computations need to be done as fixed point arithmetic. Floating point arithmetic are not implemented consistently across programming languages and architectures. This caused many problems when we implemented calculations for v2, v3.0, v3.1. I do not wish to repeat this mistake, ever.

See just a few "fun" examples here:
- https://github.com/RedHatProductSecurity/cvss/issues/10
- https://github.com/RedHatProductSecurity/cvss/issues/28
- https://github.com/RedHatProductSecurity/cvss/issues/38

Reply:
I will work on avoiding floating point arithmetic later on.